src/media_tools/media_import.c

```diff
@@ -7018,6 +7016,237 @@ static GF_Err gf_import_hevc(GF_MediaImporter *import)
+	}
+	gf_isom_set_cts_packing(import->dest, track_num, GF_TRUE);
+
+	e = gf_media_aom_parse_ivf_file_header(bs, &state);
+	if (!e) {
+		gf_import_message(import, GF_OK, "Detected IVF.");
+		av1_bs_syntax = IVF;
+		pos = gf_bs_get_position(bs);
+	} else {
+		gf_bs_seek(bs, pos);
+		e = aom_av1_parse_obu_from_annexb(bs, &state);
+		if (!e) {
+			gf_import_message(import, GF_OK, "Detected Annex B.");
+			av1_bs_syntax = AnnexB;
+		} else {
+			gf_bs_seek(bs, pos);
+			e = aom_av1_parse_obu_from_section5(bs, &state);
+			if (e) {
+				gf_import_message(import, GF_OK, "Couldn't guess bitstream format (IVF then Annex B then Section 5 tested.");
+				goto exit;
+			}
+			gf_import_message(import, GF_OK, "OBUs Section 5.");
+			av1_bs_syntax = OBUs;
+		}
+	}
+
+	gf_bs_seek(bs, pos);
+	while (gf_bs_available(bs)) {
+		av1_reset_frame_state(&state.frame_state);
+		pos = gf_bs_get_position(bs);
+
+		/*we process each TU and extract only the necessary OBUs*/
+		if (av1_bs_syntax == OBUs) {
+			if (aom_av1_parse_obu_from_section5(bs, &state) != GF_OK) {
+				gf_import_message(import, GF_OK, "Error parsing OBU (Section 5)");
+				goto exit;
+			}
+		} else if (av1_bs_syntax == AnnexB) {
+			if (aom_av1_parse_obu_from_annexb(bs, &state) != GF_OK) {
+				gf_import_message(import, GF_OK, "Error parsing OBU (Annex B)");
+				goto exit;
+			}
+		} else if (av1_bs_syntax == IVF) {
+			if (aom_av1_parse_obu_from_ivf(bs, &state) != GF_OK) {
+				gf_import_message(import, GF_OK, "Error parsing OBU (IVF)");
+				goto exit;
+			}
+		}
+
+		/*add sample*/
+		{
+			u32 i = 0;
+			GF_ISOSample *samp = gf_isom_sample_new();
+			samp->DTS = (u64)dts_inc*cur_samp;
+			samp->IsRAP = state.frame_state.key_frame ? SAP_TYPE_1 : 0;
+			samp->CTS_Offset = 0;
+
+			for (i = 0; i < gf_list_count(state.frame_state.frame_obus); ++i)
+				samp->dataLength += (u32)((GF_AV1_OBUArrayEntry*)gf_list_get(state.frame_state.frame_obus, i))->obu_length;
+			samp->data = gf_malloc(samp->dataLength);
+
+			samp->dataLength = 0;
+			while (gf_list_count(state.frame_state.frame_obus)) {
+				GF_AV1_OBUArrayEntry *a = (GF_AV1_OBUArrayEntry*)gf_list_get(state.frame_state.frame_obus, 0);
+				if (a->obu) {
+					memcpy(samp->data + samp->dataLength, a->obu, (size_t)a->obu_length);
+					samp->dataLength += (u32)a->obu_length;
+					gf_free(a->obu);
+				}
+				gf_list_rem(state.frame_state.frame_obus, 0);
+				gf_free(a);
+			}
+
+			if (cur_samp == 0) {
+				while (gf_list_count(state.frame_state.header_obus)) {
+					GF_AV1_OBUArrayEntry *a = (GF_AV1_OBUArrayEntry*)gf_list_get(state.frame_state.header_obus, 0);
+					gf_list_add(av1_cfg->obu_array, a);
+					gf_list_rem(state.frame_state.header_obus, 0);
+				}
+
+				e = gf_isom_av1_config_new(import->dest, track_num, av1_cfg, NULL, NULL, &di);
+				if (e) goto exit;
+			} else {
+				/*safety check: we only support static metadata*/
+				if (gf_list_count(state.frame_state.header_obus) > gf_list_count(av1_cfg->obu_array)) {
```

Comment:
Given that you already iterate through all frame_state.header_obus below, you don't need to do this check. Also technically, you could have repeated sequence header or metadata OBUs (weird but possible). I would remove this check.

Reply:
I'm not sure to understand. If the obus stored in the moov differ from the ones found in the first TU, then our current implementation fails. That's a limitation.